clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -173,19 +179,22 @@ CompletableFuture<Void> sendAutoCommit(final Map<TopicPartition, OffsetAndMetada
     private class OffsetCommitRequestState {
         private final Map<TopicPartition, OffsetAndMetadata> offsets;
         private final String groupId;
-        private final GroupState.Generation generation;
-        private final String groupInstanceId;
+        private final Optional<String> memberId;
+        private final Optional<String> groupInstanceId;
         private final CompletableFuture<Void> future;
+        private int memberEpoch;
```

Comment:
This should be Optional here too, right? Just as the memberId and groupInstanceId (groupId is the only required one to be able to commit offsets)